### ISSUE TITLE
On 5.8 kernels unprivileged whiteout creation is allowed

### ISSUE THREAD

**Issue by brauner** [systemd/systemd#16721]
@poettering one of your colleagues made it possible to create unprivileged whiteouts in newer kernels:

```
commit a3c751a50fe6bbe50eb7622a14b18b361804ee0c
Author: Miklos Szeredi <mszeredi@redhat.com>
Date:   Thu May 14 16:44:23 2020 +0200

    vfs: allow unprivileged whiteout creation

    Whiteouts, unlike real device node should not require privileges to create.

    The general concern with device nodes is that opening them can have side
    effects.  The kernel already avoids zero major (see
    Documentation/admin-guide/devices.txt).  To be on the safe side the patch
    explicitly forbids registering a char device with 0/0 number (see
    cdev_add()).

    This guarantees that a non-O_PATH open on a whiteout will fail with ENODEV;
    i.e. it won't have any side effect.

    Signed-off-by: Miklos Szeredi <mszeredi@redhat.com>
```

This means:

https://github.com/systemd/systemd/blob/830171936c4e9c072e85685d8c061175dda2ad71/src/test/test-fs-util.c#L620

should likely be adapted to test some other device. We've seen failures for this in our test-suites.

**Comment by brauner** [comment]
Probably also need to change:
https://github.com/systemd/systemd/blob/28b05d1e6ece19cd83afaf8f12f3fc5418151a29/src/test/test-stat-util.c#L151
because `inaccessiable/{chr,blk}` are whiteouts as well afaict.

**Comment by poettering** [comment]
it would probably suffice to swap the order of the char vs. block device node testing, right? since block devices of zero major are still not allowed, right? it's only the char devices 0:0 that are now allowed, correct?

**Comment by sforshee** [comment]
Yes, it's only the 0:0 char device that is now allowed, so testing the block device node first should be sufficient.

**Comment by poettering** [comment]
fix pending in #16734
